test/python/pulse/test_block.py

```diff
@@ -96,11 +96,8 @@ def test_equispace_alignment(self):
 
         ref_sched = pulse.Schedule()
         ref_sched = ref_sched.insert(0, pulse.Play(self.test_waveform0, self.d0))
-        ref_sched = ref_sched.insert(100, pulse.Delay(200, self.d0))
```

Comment:
I know there are more unittest updates coming, but what is this change for? Do these fail?

Reply:
Previously, delay is inserted only for `align_equispaced` and `align_func` because they have a duration for the context, but sometime the schedule duration becomes shorter than that. For example,
```python
with align_equispaced(duration=1000):
    play(Gaussian(200, 0.1, 50), d0)
```
The returned schedule is
```
(400, Play(Gaussian(200, 0.1, 50), DriveChannel(0)))
```
Note that we need to insert delay otherwise following instruction may interrupt current context.

In this PR, the behavior is updated so that inserting barrier instead of delay, at transformation phase. So inserting unintended delays is no longer necessary.
https://github.com/nkanazawa1989/qiskit-terra/blob/53cd6bb2c7e6f2118543335091cdddbd4c491271/qiskit/pulse/transforms/canonicalization.py#L66-L68